pkg/sql/sem/builtins/pg_builtins.go

```diff
@@ -1145,6 +1161,91 @@ SELECT description
 		},
 	),
 
+	"pg_relation_is_updatable": makeBuiltin(
+		defProps(),
+		tree.Overload{
+			Types:      tree.ArgTypes{{"reloid", types.Oid}, {"include_triggers", types.Bool}},
+			ReturnType: tree.FixedReturnType(types.Int4),
+			Fn: func(ctx *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				oidArg := tree.MustBeDOid(args[0])
+				oid := int(oidArg.DInt)
+				table, err := ctx.Planner.GetImmutableTableInterfaceByID(ctx.Ctx(), oid)
+				if err != nil {
+					// For postgres compatibility, it is expected that rather returning
+					// an error this return nonUpdatableEvents (Zero) because there could
+					// be oid references on deleted tables.
+					if sqlerrors.IsUndefinedRelationError(err) {
+						return nonUpdatableEvents, nil
+					}
+					return nonUpdatableEvents, err
+				}
+				tableDesc, ok := table.(catalog.TableDescriptor)
+				if !ok || !tableDesc.IsTable() || tableDesc.IsVirtualTable() {
+					return nonUpdatableEvents, nil
+				}
+
+				// pg_relation_is_updatable was created for compatibility. This
+				// should return the update events the relation supports, but as crdb
+				// does not support updatable views or foreign tables, right now this
+				// basically return allEvents or none.
+				return allUpdatableEvents, nil
+			},
+			Info:       `Returns the update events the relation supports.`,
+			Volatility: tree.VolatilityStable,
+		},
+	),
+
+	"pg_column_is_updatable": makeBuiltin(
+		defProps(),
+		tree.Overload{
+			Types: tree.ArgTypes{
+				{"reloid", types.Oid},
+				{"attnum", types.Int2},
+				{"include_triggers", types.Bool},
+			},
+			ReturnType: tree.FixedReturnType(types.Bool),
+			Fn: func(ctx *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				oidArg := tree.MustBeDOid(args[0])
+				attNumArg := tree.MustBeDInt(args[1])
+				oid := int(oidArg.DInt)
+				attNum := uint32(attNumArg)
+				if attNumArg < 0 {
+					// system columns are not updatable
+					return tree.DBoolFalse, nil
+				}
+				table, err := ctx.Planner.GetImmutableTableInterfaceByID(ctx.Ctx(), oid)
+				if err != nil {
+					if sqlerrors.IsUndefinedRelationError(err) {
+						// For postgres compatibility, it is expected that rather returning
+						// an error this return nonUpdatableEvents (Zero) because there could
+						// be oid references on deleted tables.
+						return tree.DBoolFalse, nil
+					}
+					return tree.DBoolFalse, err
+				}
+				tableDesc, ok := table.(catalog.TableDescriptor)
+				if !ok || !tableDesc.IsTable() || tableDesc.IsVirtualTable() {
+					return tree.DBoolFalse, nil
+				}
+
+				column, err := tableDesc.FindColumnWithID(descpb.ColumnID(attNum))
+				if err != nil {
+					if sqlerrors.IsUndefinedColumnError(err) {
+						return tree.DBoolFalse, nil
```

Comment:
are you sure this is false? I get true. In PostgreSQL:

```
otan=# create table new_table(id int primary key);
CREATE TABLE
otan=# select pg_column_is_updatable('new_table'::regclass::oid, 3842::int2, false);
 pg_column_is_updatable
------------------------
 t
(1 row)
```

Reply:
Nice catch, you are right!